guard/tests/validate.rs

```diff
@@ -328,44 +365,425 @@ mod validate_tests {
+        let status_code = ValidateTestRunner::default()
+            .payload()
+            .previous_engine(true)
+            .verbose(true)
+            .run(&mut writer, &mut reader);
+
+        let result = writer.stripped().unwrap();
+        let expected = indoc! {
+            r#"
+            Evaluation Tree
+            Rule(default, PASS)
+                |  Message: DEFAULT MESSAGE(PASS)
+                Clause( Parameters.InstanceName EQUALS  "TestInstance", PASS)
+                    |  Message: DEFAULT MESSAGE(PASS)
+            Evaluation Tree
+            Rule(default, PASS)
+                |  Message: DEFAULT MESSAGE(PASS)
+                Clause( Parameters.InstanceName EQUALS  "TestInstance", PASS)
+                    |  Message: DEFAULT MESSAGE(PASS)
+            DATA_STDIN[1] Status = FAIL
+            FAILED rules
+            RULES_STDIN[2]/default    FAIL
+            ---
+            Evaluation Tree
+            Rule(default, FAIL)
+                |  Message: DEFAULT MESSAGE(FAIL)
+                Clause( Parameters.InstanceName EQUALS  "SomeRandomString", FAIL)
+                    |  From: String((Path("/Parameters/InstanceName", Location { line: 0, col: 276 }), "TestInstance"))
+                    |  To: String((Path("", Location { line: 0, col: 0 }), "SomeRandomString"))
+                    |  Message: DEFAULT MESSAGE(FAIL)
+            DATA_STDIN[2] Status = FAIL
+            FAILED rules
+            RULES_STDIN[2]/default    FAIL
+            ---
+            Evaluation Tree
+            Rule(default, FAIL)
+                |  Message: DEFAULT MESSAGE(FAIL)
+                Clause( Parameters.InstanceName EQUALS  "SomeRandomString", FAIL)
+                    |  From: String((Path("/Parameters/InstanceName", Location { line: 0, col: 276 }), "TestInstance"))
+                    |  To: String((Path("", Location { line: 0, col: 0 }), "SomeRandomString"))
+                    |  Message: DEFAULT MESSAGE(FAIL)
+            "#
+        };
+
+        assert_eq!(StatusCode::PARSING_ERROR, status_code);
+        assert_eq!(expected, result);
+    }
+
+    #[test]
+    fn test_rules_with_data_from_stdin_fail_prev_engine_show_clause_failures() {
+        let path = "payload_verbose_show_failure.out";
+        let file = File::open(
+            "resources/validate/data-dir/s3-public-read-prohibited-template-non-compliant.yaml",
+        )
+        .expect("failed to find mocked file ");
+        let mut reader = Reader::new(ReadFile(file));
+        let mut writer = Writer::new(WBVec(vec![]), WBVec(vec![]));
+
+        let status_code = ValidateTestRunner::default()
+            .rules(vec!["rules-dir/s3_bucket_public_read_prohibited.guard"])
+            .show_summary(vec!["none", "all", "pass", "fail", "skip"])
+            .verbose(true)
+            .previous_engine(true)
+            .show_clause_failures(true)
+            .run(&mut writer, &mut reader);
+
+        let mut file =
```

Comment:
We should use `assert_output_from_file_eq` macro here, where we compare a buffer with file.